src/Snake.java

```diff
@@ -0,0 +1,272 @@
+        this.setPosition(SnakeGame.CANVAS_WIDTH * 0.5, SnakeGame.CANVAS_HEIGHT * 0.5);
+    }
+
+    /**
+     * Creates a list of points (path) 2.5 units behind the snake's head (in the diretion it is currently going), 
+     * with adjusting the distance of the points away from a line directly behind the snake on intervals of four.
+     * This way, every four points is allowed to be a new distance away from the line directly behind the snake's
+     * head, allowing for an object following along said path (consitently 
+     * setting its position to the next point on the list) to appear that far away from the said line. 
+     * This is adjusted so that the path appears in a curved pattern, simulating a snake slithering.
+     */
+
+    public void addToPath(List<Point> path) {
+        Point point = new Point(this.getX(), this.getY());
+        if (movingLeft) {
+            if ((i % 32) < 4) {
+                point = new Point(this.getX() + 2.5, this.getY());
+            } else if ((i % 32) >= 4 && (i % 32) < 8) {
+                point = new Point(this.getX() + 2.5, this.getY() + 1.25);
+            } else if ((i % 32) >= 8 && (i % 32) < 12) {
+                point = new Point(this.getX() + 2.5, this.getY() + 2.5);
+            } else if ((i % 32) >= 12 && (i % 32) < 16) {
+                point = new Point(this.getX() + 2.5, this.getY() + 1.25);
+            } else if ((i % 32) >= 16 && (i % 32) < 20) {
+                point = new Point(this.getX() + 2.5, this.getY());
+            } else if ((i % 32) >= 20 && (i % 32) < 24) {
+                point = new Point(this.getX() + 2.5, this.getY() - 1.25);
+            } else if ((i % 32) >= 24 && (i % 32) < 28) {
+                point = new Point(this.getX() + 2.5, this.getY() - 2.5);
+            } else if ((i % 32) >= 28 && (i % 32) < 32) {
+                point = new Point(this.getX() + 2.5, this.getY() - 1.25);
+            }
+        } else if (movingRight) {
+            if ((i % 32) < 4) {
+                point = new Point(this.getX() - 2.5, this.getY());
+            } else if ((i % 32) >= 4 && (i % 32) < 8) {
+                point = new Point(this.getX() - 2.5, this.getY() + 1.25);
+            } else if ((i % 32) >= 8 && (i % 32) < 12) {
+                point = new Point(this.getX() - 2.5, this.getY() + 2.5);
+            } else if ((i % 32) >= 12 && (i % 32) < 16) {
+                point = new Point(this.getX() - 2.5, this.getY() + 1.25);
+            } else if ((i % 32) >= 16 && (i % 32) < 20) {
+                point = new Point(this.getX() - 2.5, this.getY());
+            } else if ((i % 32) >= 20 && (i % 32) < 24) {
+                point = new Point(this.getX() - 2.5, this.getY() - 1.25);
+            } else if ((i % 32) >= 24 && (i % 32) < 28) {
+                point = new Point(this.getX() - 2.5, this.getY() - 2.5);
+            } else if ((i % 32) >= 28 && (i % 32) < 32) {
+                point = new Point(this.getX() - 2.5, this.getY() - 1.25);
+            }
```

Comment:
The enormous degree of repetition in this code cries out for refactoring! I’m pretty sure you could fold these ~70 lines into about 6 by finding general formulas for the various directions and motion phases instead of making everything a special case. Still, good job figuring it out.